### PR TITLE
ipsec: do not explicitly prioritize host endpoint regeneration

### PR DESCRIPTION
**Pull request by ldelossa**
This code was introduced in #25735.

As described well in:

16e446c0ffd6424a1a1b737cb359c6e9ef339e3d
bpf: Support the old IP_POOLS logic in bpf_host

and

b429c42a4f8d7217cc998ffa50d0d18d9e27054b
daemon: Reload bpf_host first in case of IPsec upgrade

This code was put in place to ensure the refactored 'bpf_host' program, which supports the IP_POOLS hack upgrade compatibility, is installed before any other eBPF prog.

Now that we are N+2 versions from this introduction, and we instruct individuals to upgrade consecutively, this code can be removed.

It is possible that something in Cilium now depends on this eBPF program installation ordering. The status of CI jobs in this PR should be looked at with a keener-then-normal eye due to this possibility. 

```release-note
No longer prioritizes HostEndpoint regeneration during Cilium init. 
```

### CODE REVIEW

**Review by pchaigno — APPROVED** [review]

File: daemon/cmd/state.go [hunk 1]
```diff
@@ -307,42 +307,7 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState, endpoi
...
-		// In newer versions, the header is written by the kernel XFRM
-		// subsystem and bpf_host must therefore not write it. To allow for a
-		// smooth upgrade, bpf_host has been updated to handle both cases. But
-		// for that to succeed, it must be reloaded first, before the bpf_lxc
-		// programs stop writing the IP into skb->cb.
-		for _, ep := range state.restored {
-			// Cap the timeout used to wait for remote cluster synchronization
-			// to avoid blocking the agent startup, as this regeneration is
-			// performed synchronously.
-			endpointsRegenerator.CapTimeoutForSynchronousRegeneration()
```
Comment (giorio94) [inline daemon/cmd/state.go:326]: Drive-by comment, please delete this function as well.

Comment (ldelossa) [reply daemon/cmd/state.go:326]: Good call, you beat golangci-lint :-D

Comment (giorio94) [reply daemon/cmd/state.go:326]: Just waiting for the opportunity to remove that hack I had to introduce a while ago for that specific case 😁

**Review by squeed — APPROVED** [review]
